packages/styled-components/src/models/StyledComponent.js

```diff
@@ -277,15 +277,18 @@ export default function createStyledComponent(target: Target, options: Object, r
    * forwardRef creates a new interim component, which we'll take advantage of
    * instead of extending ParentComponent to create _another_ interim class
    */
-  const WrappedStyledComponent = React.forwardRef((props, ref) => (
+  let WrappedStyledComponent;
+  const forwardRef = (props, ref) => (
     <ParentComponent {...props} forwardedComponent={WrappedStyledComponent} forwardedRef={ref} />
-  ));
+  );
+  forwardRef.displayName = displayName;
+  WrappedStyledComponent = React.forwardRef(forwardRef);
+  WrappedStyledComponent.displayName = displayName;
```

Comment:
I guess this one isn't needed if you put it on the underlying one?

Reply:
Yeah but all tests fail otherwise, will need to understand how the tests check this